docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -1248,94 +1348,19 @@ Locking and unlocking of owned objects are two crucial on-chain actions in this
 
-			txb.transferObjects([locked, key], txb.pure.address(account.address));
+{@inject: examples/trading/frontend/src/mutations/locked.ts#mutationunlock}
 
-			return executeTransaction(txb);
-		},
-	});
-}
-```
+#### Finished locked mutations
 
-#### Unlock owned objects
+When finished, the `locked.ts` file should look like this:
 
-To unlock the object, execute the `unlock` Move function identified by `{PACKAGE_ID}::lock::unlock`. The implementation is straightforward, call the `unlock` function supplying the `Locked` object, its corresponding `Key`, the struct type of the original object, and transfer the unlocked object to the current connected wallet account. Also, implement the `onSuccess` callback to invalidate the cache data at query key `locked` after one second to force `react-query` to re-fetch the data at corresponding query key automatically.
-
-```ts title='src/mutations/locked.ts'
-export function useUnlockMutation() {
-	const account = useCurrentAccount();
-	const executeTransaction = useTransactionExecution();
-	const client = useSuiClient();
-	const queryClient = useQueryClient();
-
-	return useMutation({
-		mutationFn: async ({
-			lockedId,
-			keyId,
-			suiObject,
-		}: {
-			lockedId: string;
-			keyId: string;
-			suiObject: SuiObjectData;
-		}) => {
-			if (!account?.address) throw new Error('You need to connect your wallet!');
-			const key = await client.getObject({
-				id: keyId,
-				options: {
-					showOwner: true,
-				},
-			});
-
-			if (
-				!key.data?.owner ||
-				typeof key.data.owner === 'string' ||
-				!('AddressOwner' in key.data.owner) ||
-				key.data.owner.AddressOwner !== account.address
-			) {
-				toast.error('You are not the owner of the key');
-				return;
-			}
-
-			const txb = new Transaction();
-
-			const item = txb.moveCall({
-				target: `${CONSTANTS.escrowContract.packageId}::lock::unlock`,
-				typeArguments: [suiObject.type!],
-				arguments: [txb.object(lockedId), txb.object(keyId)],
-			});
-
-			txb.transferObjects([item], txb.pure.address(account.address));
-
-			return executeTransaction(txb);
-		},
-		onSuccess: () => {
-			setTimeout(() => {
-				// invalidating the queries after a small latency
-				// because the indexer works in intervals of 1s.
-				// if we invalidate too early, we might not get the latest state.
-				queryClient.invalidateQueries({
-					queryKey: [QueryKey.Locked],
-				});
-			}, 1_000);
-		},
-	});
-}
-```
+{@inject: examples/trading/frontend/src/mutations/locked.ts}
```

Comment:
Just a note. We have the option of hiding long codeblocks inside expandable text. I don't think it's necessary for these guides but others might disagree. The format would be:
```
<details>
<summary>
Toggle code source
</summary>
{@inject: ...}
</details> 
```